File: test/Graphics/RectangleInstructionsTest.java
package Graphics;

import Models.Measurement;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Vector;

import static org.junit.jupiter.api.Assertions.*;

class RectangleInstructionsTest {
    Measurement zero = new Measurement(0);
    Measurement one = new Measurement(1);
    Measurement ten = new Measurement(10);
    @Test
    public void shouldImplementDrawingInstructions() {
        Integer one = new Measurement(1).numberOfPixels();
        Vector<Integer> result = new Vector<>(new Vector<>(List.of(one, one, one, one)));
        Measurement oneM = new Measurement(1);
        assertEquals(result, new RectangleInstructions(oneM, oneM, oneM, oneM).drawingInstructions());
    }

    @Test
    public void shouldShift() {
        Vector<Integer> result = new Vector<>(List.of(ten.numberOfPixels(), ten.numberOfPixels(), one.numberOfPixels(), one.numberOfPixels()));
        RectangleInstructions test = new RectangleInstructions(zero, zero, one, one);
        assertNotEquals(result, test.drawingInstructions());
        assertEquals(result, test.shift(ten, ten).drawingInstructions());
    }
}

File: src/test/kotlin/graphics/RectangleInstructionsTest.kt
package graphics

import models.Measurement
import kotlin.test.*
import java.util.*

internal class RectangleInstructionsTest {
    var zero = Measurement(0)
    private var one = Measurement(1)
    private var ten = Measurement(10)

    @Test
    fun shouldImplementDrawingInstructions() {
        val one = Measurement(1).numberOfPixels
        val result = Vector(Vector(listOf(one, one, one, one)))
        val oneM = Measurement(1)
        assertEquals(result, RectangleInstructions(oneM, oneM, oneM, oneM).drawingInstructions())
    }

    @Test
    fun shouldShift() {
        val result = Vector(listOf(ten.numberOfPixels, ten.numberOfPixels, one.numberOfPixels, one.numberOfPixels))
        val test = RectangleInstructions(zero, zero, one, one)
        assertNotEquals(result, test.drawingInstructions())
        assertEquals(result, test.shift(ten, ten).drawingInstructions())
        assertEquals(result, test.shift(ten, ten).drawingInstructions())
    }
}